### PR TITLE
feat(location): adding battery level data

### PR DESCRIPTION
**Pull request by namchuai**
- Adding battery level to each location update.

### CODE REVIEW

File: SYRFLocation/SYRFLocation/Data/LocationData.swift [hunk 1]
```diff
@@ -129,6 +134,7 @@ public class SYRFLocation {
             self.courseAccuracy = 0
         }
         self.courseHeading = location.course
+        self.batteryLevel = Double(UIDevice.current.batteryLevel)
```
Comment (radurad) [inline SYRFLocation/SYRFLocation/Data/LocationData.swift:137]: Using batteryLevel directly without requesting monitoring will yield a -1.0 value always as stated here: https://developer.apple.com/documentation/uikit/uidevice/1620042-batterylevel?language=objc

You'll have to explicitly request notification or read capability for the battery as stated here: https://developer.apple.com/documentation/uikit/uidevice/1620045-batterymonitoringenabled?language=objc.

If battery level is the only new addition then we can tie it up with start/stopLocationUpdates, whenever called, enabled/disable battery monitoring. If there are other readings that we need to extract apart from the battery level, we should consider creating a dedicated manager for them as well.

**Review by radurad — CHANGES_REQUESTED** [review]

File: SYRFLocation/SYRFLocation/Data/LocationData.swift (hunk 1 above)
Comment (namchuai) [reply SYRFLocation/SYRFLocation/Data/LocationData.swift:137]: omg, Sorry @radurad , I misread that part. I will update the PR. Thanks for pointing it out!

Comment (namchuai) [reply SYRFLocation/SYRFLocation/Data/LocationData.swift:137]: @radurad , I updated the PR to add the enable/disable battery monitoring. Thanks!

**Comment by namchuai** [comment]
closing to separate module

**Pull request closed without merging**